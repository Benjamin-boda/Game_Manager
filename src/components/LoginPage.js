import React from "react";
import {useDispatch} from "react-redux";
import {startLogin} from "../actions/auth";

export const LoginPage = () => {
    const dispatch = useDispatch()
    return (
        <div className="box-layout">
            <div className="box-layout__box">
                <h2 className="box-layout__title">Connect to see your game list</h2>
                <p>Manage to finish your games</p>
                <button className="button__login" onClick={() => {dispatch(startLogin())}}>Login with Google</button>
            </div>
        </div>
    )
};

export default LoginPage;