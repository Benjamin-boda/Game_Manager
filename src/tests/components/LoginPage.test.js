import LoginPage from "../../components/LoginPage";
import React from "react";
import { shallow, mount } from "enzyme";
import * as redux from 'react-redux'
import configureMockStore from "redux-mock-store";
import { shallowToJson } from "enzyme-to-json"
import thunk from 'redux-thunk'
import {Provider} from "react-redux";

const middlewares = [thunk]
const mockStore = configureMockStore(middlewares);
const store = mockStore({});



test("Should render LoginPage component correctly", () => {
    
    const wrapper = mount(
        <Provider store={store}>
            <LoginPage />
        </Provider>
        );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
});

test("Should call startlogin on button click", () => {

    const spy = jest.spyOn(redux, 'useDispatch')
    spy.mockReturnValue(jest.fn())
    const wrapper = mount(<Provider store={store}>
            <LoginPage startLogin={spy}/>
        </Provider>
        );
    wrapper.find("button").simulate("click")
    expect(spy).toHaveBeenCalled();
});