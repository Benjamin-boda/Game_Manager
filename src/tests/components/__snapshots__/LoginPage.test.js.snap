// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render LoginPage component correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <LoginPage>
    <div>
      <div>
        <h2>
          Connect to see your game list
        </h2>
        <p>
          Manage to finish your games
        </p>
        <button
          onClick={[Function]}
        >
          Login with Google
        </button>
      </div>
    </div>
  </LoginPage>
</Provider>
`;
